ChildProject/annotations.py

```diff
@@ -0,0 +1,218 @@
+        IndexColumn(name = 'speaker_type', description = '', required = True),
+        IndexColumn(name = 'ling_type', description = '1 if the vocalization contains at least a vowel (ie canonical or non-canonical)', required = True),
+        IndexColumn(name = 'vcm_type', description = '', required = True),
+        IndexColumn(name = 'lex_type', description = '', required = True),
+        IndexColumn(name = 'mwu_type', description = '', required = True),
+        IndexColumn(name = 'addresseee', description = '', required = True),
+        IndexColumn(name = 'transcription', description = '', required = True)
+    ]
+
+    SPEAKER_ID_TO_TYPE = {
+        'C1': 'OCH',
+        'C2': 'OCH',
+        'CHI': 'CHI',
+        'CHI*': 'CHI',
+        'EE1': 'ELE',
+        'FA0': 'FEM',
+        'FA1': 'FEM',
+        'FA2': 'FEM',
+        'FA3': 'FEM',
+        'FA4': 'FEM',
+        'FA5': 'FEM',
+        'FA6': 'FEM',
+        'FA7': 'FEM',
+        'FA8': 'FEM',
+        'FAE': 'ELE',
+        'FC1': 'OCH',
+        'FC2': 'OCH',
+        'FC3': 'OCH',
+        'MA0': 'MAL',
+        'MA1': 'MAL',
+        'MA2': 'MAL',
+        'MA3': 'MAL',
+        'MA4': 'MAL',
+        'MA5': 'MAL',
+        'MAE': 'ELE',
+        'MC1': 'OCH',
+        'MC2': 'OCH',
+        'MC3': 'OCH',
+        'MC4': 'OCH',
+        'MC5': 'OCH',
+        'MI1': 'OCH',
+        'MOT*': 'FEM',
+        'OC0': 'OCH',
+        'UC1': 'OCH',
+        'UC2': 'OCH',
+        'UC3': 'OCH',
+        'UC4': 'OCH',
+        'UC5': 'OCH',
+        'UC6': 'OCH'
+    }
+
+
+    def __init__(self, project):
+        self.project = project
+        self.annotations = None
+        self.errors = []
+
+        if not isinstance(project, ChildProject):
+            raise ValueError('project should derive from ChildProject')
+
+        project.read()
+
+        index_path = os.path.join(self.project.path, 'annotations/annotations.csv')
+        if not os.path.exists(index_path):
+            open(index_path, 'w+').write(','.join([c.name for c in self.INDEX_COLUMNS]))
+
+        errors, warnings = self.read()
+
+    def read(self):
+        table = IndexTable('input', path = os.path.join(self.project.path, 'annotations/annotations.csv'), columns = self.INDEX_COLUMNS)
+        self.annotations = table.read()
+        errors, warnings = table.validate()
+        return errors, warnings
+
+    def validate(self):
+        errors, warnings = [], []
+
+        for annotation in self.annotations.to_dict(orient = 'records'):
+            segments = IndexTable(
+                'segments',
+                path = os.path.join(self.project.path, 'annotations', annotation['annotation_filename']),
+                columns = AnnotationManager.SEGMENTS_COLUMNS
+            )
+
+            segments.read()
+            res = segments.validate()
+            errors += res[0]
+            warnings += res[1]
+
+        return errors, warnings
+        
+
+    def load_textgrid(self, filename):
```

Comment:
Only one question:
SPEAKER_ID_TO_TYPE -- so we'll generate the types (line 141 in the next chunk)? that's what we did in the past with exactly the code you see there. BUT notice that this code ignores far away speech (LF2P) and overlapping speech (2POPMT), which we don't want to lose. (In the past Marvin dropped it because he was training a model, so he needed clean speech.) This also removes Noise (a tier called noise) but this was annotated in such an erratic way that we can probably forget it. (I'd also rewrite Marvin's code with regular expressions, but I know that's individual choice).
Otherwise this bit looks great.

Reply:
What do you think is the desired behavior in this case ? Right now, the information is not lost at least, but SPEAKER TYPE is NA when ID = LF2P or 2POPMT

Reply:
this is indeed the right behavior! If Marvin or someone else wants to retrain, they can select down by speaker type removing NAs